### PR TITLE
add DrawingController.cs

### PR DESCRIPTION
**Pull request by kaallaash**

### CODE REVIEW

File: DescriptiveGeometry/DG.API/Controllers/DrawingController.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+    [HttpGet("ids/q-{query}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(typeof(void), StatusCodes.Status404NotFound)]
+    public async Task<ActionResult<List<int>>> GetId(
+        [FromRoute] string query)
+    {
+        var result = await _drawingService
+            .GetId(query)
+            .ConfigureAwait(false);
```
Comment (moltenessence) [inline DescriptiveGeometry/DG.API/Controllers/DrawingController.cs:66]: What are benefits of using 'ConfigureAwait(false)' in controller?

Comment (kaallaash) [reply DescriptiveGeometry/DG.API/Controllers/DrawingController.cs]: > What are benefits of using 'ConfigureAwait(false)' in controller?
Yes. You're right. It's useless.

File: DescriptiveGeometry/DG.API/Controllers/DrawingController.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    private readonly IDrawingService _drawingService;
+
+    public DrawingController(
+        IDrawingService drawingService)
+    {
+        _drawingService = drawingService;
+    }
+
+    [HttpGet("{id}")]
+    public async Task<ActionResult<Drawing>> Get(
```
Comment (nantonov) [inline DescriptiveGeometry/DG.API/Controllers/DrawingController.cs:21]: ActionResult for what?

File: DescriptiveGeometry/DG.API/Controllers/DrawingController.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        CancellationToken cancellationToken = default)
+    {
+        var result = await _drawingService
+            .Get(id, cancellationToken);
+
+        return result;
+    }
+
+    [HttpGet("description/{description}")]
+    public async Task<List<Drawing>> FindByDescription(
```
Comment (nantonov) [inline DescriptiveGeometry/DG.API/Controllers/DrawingController.cs:32]: Why not ActionResult here?

File: DescriptiveGeometry/DG.API/Controllers/DrawingController.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        [FromRoute] string description,
```
Comment (nantonov) [inline DescriptiveGeometry/DG.API/Controllers/DrawingController.cs:33]: Why FromRoute?

File: DescriptiveGeometry/DG.API/Controllers/DrawingController.cs [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        CancellationToken cancellationToken = default)
+    {
+        var result = await _drawingService
+            .GetByQuery(description, cancellationToken);
+
+        return result;
+    }
+
+    [HttpGet]
+    public async Task<List<Drawing>> GetDrawings(
```
Comment (nantonov) [inline DescriptiveGeometry/DG.API/Controllers/DrawingController.cs:43]: Why in the name you use Drawings here?
But in the other endpoints you don't use it

File: DescriptiveGeometry/DG.API/Controllers/DrawingController.cs [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        CancellationToken cancellationToken = default)
+    {
+        var result = await _drawingService
+            .GetDrawings(cancellationToken);
+
+        return result;
+    }
+
+    [HttpGet("ids/description/{description}")]
+    public async Task<List<int>> GetId(
```
Comment (nantonov) [inline DescriptiveGeometry/DG.API/Controllers/DrawingController.cs:53]: What is the purpose of this endpoint?

Comment (nantonov) [reply DescriptiveGeometry/DG.API/Controllers/DrawingController.cs:53]: Name doesn't reflect the logic of this endpoint

File: DescriptiveGeometry/DG.API/Controllers/DrawingController.cs [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    {
+        var result = await _drawingService
+            .Create(drawing, cancellationToken);
+
+        return result;
+    }
+
+    [HttpPut]
+    public async Task<int> Update(
+      [FromBody] Drawing drawing,
```
Comment (nantonov) [inline DescriptiveGeometry/DG.API/Controllers/DrawingController.cs:76]: In RESTful API you mustn't pass ids through body in PUT method

**Review by nantonov — CHANGES_REQUESTED** [review]

File: DescriptiveGeometry/DG.API/Controllers/DrawingController.cs (hunk 4 above)
Comment (kaallaash) [reply DescriptiveGeometry/DG.API/Controllers/DrawingController.cs]: > Why FromRoute?
Edit on FromQuery

File: DescriptiveGeometry/DG.API/Controllers/DrawingController.cs [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+
+    [HttpDelete]
+    public Task Delete(
+        [FromQuery] int id,
+        CancellationToken cancellationToken = default)
+    {
+        return _drawingService.Delete(id, cancellationToken);
+    }
+
+    private static Drawing MapToDrawing(ShortDrawingViewModel drawing)
```
Comment (moltenessence) [inline DescriptiveGeometry/DG.API/Controllers/DrawingController.cs:89]: Try to use automapper

File: DescriptiveGeometry/DG.API/Controllers/DrawingController.cs [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+        CancellationToken cancellationToken = default)
+    {
+        var result = await _drawingService
+            .GetByDescription(description, cancellationToken);
+
+        return result;
+    }
+    
+    [HttpGet]
+    public async Task<List<Drawing>> GetAll(
```
Comment (moltenessence) [inline DescriptiveGeometry/DG.API/Controllers/DrawingController.cs:44]: Return ViewModel instead model from BLL

File: DescriptiveGeometry/DG.API/Controllers/DrawingController.cs [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+
+        return result;
+    }
+
+    [HttpDelete]
+    public Task Delete(
+        [FromQuery] int id,
+        CancellationToken cancellationToken = default)
+    {
+        return _drawingService.Delete(id, cancellationToken);
```
Comment (moltenessence) [inline DescriptiveGeometry/DG.API/Controllers/DrawingController.cs:86]: Why this method is not async ?

**Review by moltenessence — CHANGES_REQUESTED** [review]

File: DescriptiveGeometry/DG.API/Controllers/DrawingController.cs [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+        CancellationToken cancellationToken = default)
+    {
+        var result = await _drawingService
+            .Get(id, cancellationToken);
+
+        return result;
+    }
+    
+    [HttpGet("description")]
+    public async Task<List<Drawing>> FindByDescription(
```
Comment (nantonov) [inline DescriptiveGeometry/DG.API/Controllers/DrawingController.cs:33]: FindBy or GetBy?

File: DescriptiveGeometry/DG.API/Controllers/DrawingController.cs (hunk 8 above)
Comment (kaallaash) [reply DescriptiveGeometry/DG.API/Controllers/DrawingController.cs:89]: It'll be in next PR

**Review by nantonov — APPROVED** [review]

File: DescriptiveGeometry/DG.API/Controllers/DrawingController.cs (hunk 9 above)
Comment (GodlyCat) [reply DescriptiveGeometry/DG.API/Controllers/DrawingController.cs:44]: It'll be in next PR

File: DescriptiveGeometry/DG.API/Controllers/DrawingController.cs (hunk 10 above)
Comment (GodlyCat) [reply DescriptiveGeometry/DG.API/Controllers/DrawingController.cs:86]: When we have only one async operation in async method, we could just return this async operation without async/await

**Review by GodlyCat — APPROVED** [review]

**Pull request merged**